tests/s3/versioning/test_get_put_object_tagging.py

```diff
@@ -695,3 +695,237 @@ def test_get_put_obj_tags_susp_bkt_40432(self):
+                                             version_id=latest_ver_id)
+        assert_utils.assert_true(resp[0], resp)
+        get_tag = resp[1][0]
+        assert_utils.assert_equal(get_tag, put_tag, "Mismatch in tag Key-Value pair."
+                                                    f"Expected: {put_tag} \n Actual: {get_tag}")
+
+        LOGGER.info("Step 8: Perform GET Object Tagging for %s without versionId specified",
+                    self.object_name)
+        resp = s3_cmn_lib.get_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name)
+        assert_utils.assert_false(resp[0], resp)
+        assert_utils.assert_in("NoSuchKey", resp[1].message)
+
+        LOGGER.info("Step 9: Perform GET Object Tagging for %s with deletemarkerid=%s",
+                    self.object_name, dm_id)
+        resp = s3_cmn_lib.get_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name,
+                                             version_id=dm_id)
+        assert_utils.assert_false(resp[0], resp)
+        assert_utils.assert_in("NoSuchKey", resp[1].message)
+
+        LOGGER.info("Step 10: Perform PUT Object Tagging for %s with a tag key-value pair"
+                    " with versionId=%s", self.object_name, latest_ver_id)
+        resp = s3_cmn_lib.put_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name,
+                                             version_tag=self.ver_tag, versions_dict=self.versions,
+                                             version_id=latest_ver_id)
+        assert_utils.assert_true(resp[0], resp)
+
+        LOGGER.info("Step 11: Perform GET Object Tagging for %s with versionId=%s",
+                    self.object_name, latest_ver_id)
+        put_tag = self.ver_tag[self.object_name][latest_ver_id][-1]
+        resp = s3_cmn_lib.get_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name,
+                                             version_id=latest_ver_id)
+        assert_utils.assert_true(resp[0], resp)
+        get_tag = resp[1][0]
+        assert_utils.assert_equal(get_tag, put_tag, "Mismatch in tag Key-Value pair."
```

Comment:
do we want to get put_tag again in step 10?

Reply:
@sonalk0209 this will verify PUT Object Tagging for version when delete marker is present. cc @rahul-hatwar

Reply:
Yes. We need to fetch the latest put_tag for latest_ver_id.
Otherwise, it won't have new tag key-val as per new PUT in step 10